portal-api/src/v1/controllers/countries.controller.js

```diff
@@ -1,44 +1,122 @@
 const utils = require('../../utils/array');
-
 const externalApi = require('../../external-api/api');
 
-const getCountryFromArray = (arr, code) => arr.filter((country) => country.code === code)[0];
+/**
+ * Retrieves a country object from an array based on the provided country code.
+ *
+ * @param {Array} arr - The array of country objects to search in.
+ * @param {string} code - The country code to search for.
+ * @returns {Array} - The country object matching the provided code, or an empty array if not found.
+ */
+const getCountryFromArray = (arr, code) => (arr.length ? arr.filter((country) => country.code === code)[0] : []);
```

Comment:
Just a side note -- I feel that the function here is actually belongs more in a service than a controller -- but I'm not worried for this refactor, it's more just for new things (we shall slowly refactor these over time)

I generally use the "controller handles what response is returned, and calls the right services in the right order. Services do the actual work (calling repos/external apis as needed)"

Reply:
This appears to be a bigger PR on a more standardised pattern.